internal/controllers/ipaddressclaim_test.go

```diff
@@ -1057,6 +1057,249 @@ var _ = Describe("IPAddressClaimReconciler", func() {
+				Eventually(Get(&pool)).Should(Succeed())
+
+				expectedIPAddress = &ipamv1.IPAddress{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:       existingAddressName,
+						Namespace:  namespace,
+						Finalizers: []string{ipamutil.ProtectAddressFinalizer},
+						OwnerReferences: []metav1.OwnerReference{
+							{
+								APIVersion:         "ipam.cluster.x-k8s.io/v1beta1",
+								BlockOwnerDeletion: ptr.To(true),
+								Controller:         ptr.To(true),
+								Kind:               "IPAddressClaim",
+								Name:               existingAddressName,
+							},
+							{
+								APIVersion:         "ipam.cluster.x-k8s.io/v1alpha2",
+								BlockOwnerDeletion: ptr.To(true),
+								Controller:         ptr.To(false),
+								Kind:               "InClusterIPPool",
+								Name:               poolName,
+							},
+						},
+					},
+					Spec: ipamv1.IPAddressSpec{
+						ClaimRef: corev1.LocalObjectReference{
+							Name: existingAddressName,
+						},
+						PoolRef: corev1.TypedLocalObjectReference{
+							APIGroup: ptr.To("ipam.cluster.x-k8s.io"),
+							Kind:     "InClusterIPPool",
+							Name:     poolName,
+						},
+						Address: "10.0.0.1",
+						Prefix:  24,
+						Gateway: "10.0.0.2",
+					},
+				}
+
+				claim := newClaim(existingAddressName, namespace, "InClusterIPPool", poolName)
+				Expect(k8sClient.Create(context.Background(), &claim)).To(Succeed())
+
+				Eventually(findAddress(existingAddressName, namespace)).
+					WithTimeout(time.Second).WithPolling(100 * time.Millisecond).Should(
+					EqualObject(expectedIPAddress, IgnoreAutogeneratedMetadata, IgnoreUIDsOnIPAddress),
+				)
+				addresses := ipamv1.IPAddressList{}
+				Expect(ObjectList(&addresses)()).To(HaveField("Items", HaveLen(1)))
+			})
+
+			AfterEach(func() {
+				deleteClaimUnchecked(existingAddressName, namespace)
+				deleteClaim(newAddressName, namespace)
+				deleteNamespacedPool(poolName, namespace)
+			})
+
+			It("should not allocate an Address from the Pool", func() {
+				claim := newClaim(newAddressName, namespace, "InClusterIPPool", poolName)
+
+				Expect(k8sClient.Create(context.Background(), &claim)).To(Succeed())
+
+				addresses := ipamv1.IPAddressList{}
+				Consistently(ObjectList(&addresses)).
+					WithTimeout(10 * time.Second).WithPolling(100 * time.Millisecond).Should(
+					HaveField("Items", HaveLen(1)))
+			})
+
+			It("should allow new claims to progress once capacity becomes available", func() {
+				claim := newClaim(newAddressName, namespace, "InClusterIPPool", poolName)
+
+				Expect(k8sClient.Create(context.Background(), &claim)).To(Succeed())
+
+				addresses := ipamv1.IPAddressList{}
+				Consistently(ObjectList(&addresses)).
+					WithTimeout(10 * time.Second).WithPolling(100 * time.Millisecond).Should(
+					HaveField("Items", HaveLen(1)))
+
+				// Sleep 5 seconds to give the reconciler time to stop retrying
+				// on the newly created claim.
+				time.Sleep(5 * time.Second)
```

Comment:
not ideal, but I guess there is no other way

Reply:
Yeah I didn't want to add this either but it wouldn't reproduce the issue otherwise.